sos/report/plugins/ceph_podman.py

```diff
@@ -0,0 +1,110 @@
+
+    def setup(self):
+        self._container_mon_name = self.get_container_by_name('ceph-mon-*')
+        self._container_osd_name = self.get_container_by_name('ceph-osd-*')
+
+        self.add_copy_spec([
+            "/etc/ceph/",
+            "/var/lib/ceph/",
+            "/run/ceph/"
+        ])
+
+        if self._container_mon_name:
+            cmds = [
+                'ceph mon stat',
+                'ceph mon_status',
+                'ceph quorum_status',
+                'ceph mgr module ls',
+                'ceph mgr metadata',
+                'ceph osd metadata',
+                'ceph osd erasure-code-profile ls',
+                'ceph report',
+                'ceph osd crush show-tunables',
+                'ceph versions',
+                'ceph features',
+                'ceph osd crush dump',
+                'ceph -v',
+                'ceph crash stat',
+                'ceph crash ls',
+                'ceph config log',
+                'ceph config generate-minimal-conf',
+                'ceph config-key dump',
+                'ceph status',
+                'ceph health detail',
+                'ceph osd tree',
+                "ceph osd stat",
+                "ceph osd df tree",
+                "ceph osd dump",
+                "ceph osd df",
+                "ceph osd perf",
+                "ceph osd blocked-by",
+                "ceph osd pool ls detail",
+                "ceph osd numa-status",
+                "ceph device ls",
+                "ceph mon dump",
+                "ceph mgr dump",
+                "ceph mds stat",
+                "ceph df",
+                "ceph df detail",
+                "ceph fs ls",
+                "ceph fs dump",
+                "ceph pg dump",
+                "ceph pg stat",
+            ]
+            container_cmds = self.to_container_cmd(cmds,
+                                                   self._container_mon_name)
+            self.add_cmd_output(container_cmds, foreground=True)
+
+        if self._container_osd_name:
+            cmds = [
+                'ceph -v',
+                'ceph-volume lvm list',
+            ]
+            osd_containers = self.get_all_containers_by_name('ceph-osd-*')
+            for c in osd_containers:
+                container_cmds = self.to_container_cmd(cmds, c)
+                self.add_cmd_output(container_cmds, foreground=True)
+
+        self.add_forbidden_path([
+            "/etc/ceph/*keyring*",
+            "/var/lib/ceph/*keyring*",
+            "/var/lib/ceph/*/*keyring*",
+            "/var/lib/ceph/*/*/*keyring*",
+            "/var/lib/ceph/osd",
+            "/var/lib/ceph/mon",
+            "/var/lib/ceph/tmp/*mnt*",
+            "/etc/ceph/*bindpass*"
+        ])
+
+    def to_container_cmd(self, cmds, container_name):
+        container_cmds = []
+        for cmd in cmds:
+            container_cmds.append('podman exec %s %s' % (container_name, cmd))
+        return container_cmds
```

Comment:
There is already a `Policy().format_container_cmd()` method that does this. We could also expose this via a wrapper in `Plugin` if need be.

Reply:
I found fmt_container_cmd() will return a command with quote: 

    def fmt_container_cmd(self, container, cmd):
        return "%s %s %s" % (self.run_cmd, container, quote(cmd))

But podman is not working with a quote: 

For example: 
~~~
[root@overcloud-controller-0 tmp]# podman exec  ceph-mon-overcoud-controller-0 'ceph mon stat'
Error: exec failed: container_linux.go:346: starting container process caused "exec: \"ceph mon stat\": executable file not found in $PATH": OCI runtime command not found error
[root@overcloud-controller-0 tmp]# podman exec  ceph-mon-overcoud-controller-0 ceph mon stat
e1: 3 mons at {overcloud-controller-0=[v2:172.24.20.69:3300/0,v1:172.24.20.69:6789/0],overcloud-controller-1=[v2:172.24.20.97:3300/0,v1:172.24.20.97:6789/0],overcloud-controller-2=[v2:172.24.20.49:3300/0,v1:172.24.20.49:6789/0]}, election epoch 34, leader 0 overcloud-controller-2, quorum 0,1,2 overcloud-controller-2,overcloud-controller-0,overcloud-controller-1
~~~

I'm not sure which one is proper, re-write fmt_container_cmd()  or override this function in ceph_podman.

Reply:
Seems we have a schism as there is both a `format_container_command()` used by `collect`, and then `fmt_container_cmd()` used by plugin. 

I believe the right thing to do here would be to move the former into `ContainerRuntime`, replacing the latter. We can still expose it via `Policy` as well.